clients/admin-ui/src/features/dataset/EditCollectionDrawer.tsx

```diff
@@ -30,7 +31,11 @@ const EditCollectionDrawer = ({
   isOpen,
   onClose,
 }: Props) => {
-  const collectionIndex = dataset?.collections.indexOf(collection!);
+  const collectionIndex = useMemo(
+    () => dataset?.collections.indexOf(collection),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
i'm curious, did it cause an issue trying to add the deps?

Reply:
This was causing a test to fail and I believe it's because the supplied collection was no longer matching anything in the dataset. Since we're now requiring both dataset and collection to be available from the start, we want to snag this index and then stop worrying about it once either gets updated. Including the dependencies recreates the problem.

Reply:
I think if I were to rewrite this, I'd pass in the `index` rather than the `collection` but this seemed like the less risky fix.